src/server.js

```diff
@@ -170,103 +174,195 @@ api.post('/getChunkedData', (req, res) => {
   }
   console.log(`dataPath = ${dataPath}`);
 
-  // call 'vg chunk' to generate graph
-  let vgChunkParams = ['chunk', '-x', `${dataPath}${xgFile}`];
-  if (req.withGam) {
-    // Use a GAM index
-    vgChunkParams.push('-a', `${dataPath}${gamFile}`, '-g');
+  // parse region
+  // either a range -> seq:start-end
+  // or a position and distance -> seq:post+distance
+  let region = req.body.region;
+  let region_col = region.split(":");
+  let start_end = region_col[1].split("-");
+  let pos_dist = region_col[1].split("+");
+  let r_start = -1;
+  let r_end = -1;
+  let distance = -1;
+  if(start_end.length == 2){
+    r_start = Number(start_end[0]);
+    r_end = Number(start_end[1]);
+  } else if(pos_dist.length == 2){
+    r_start = Number(pos_dist[0]);
+    distance = Number(pos_dist[1]);
   }
-  if (req.withGbwt) {
-    // Use a GBWT haplotype database
-    vgChunkParams.push('--gbwt-name', `${dataPath}${gbwtFile}`);
+
+  // check the bed file if this region has been pre-fetched
+  let chunkPath = '';
+  if(req.withBed){
+    let i = 0;
+    let nb_regions_in_bed = 0;
+    if(req.body.regionInfo && Object.keys(req.body.regionInfo).length > 0){
+      nb_regions_in_bed = req.body.regionInfo['desc'].length;
+    }
+    while (i < nb_regions_in_bed && chunkPath === ''){
+      let region_chr = req.body.regionInfo['chr'][i];
+      let region_start = req.body.regionInfo['start'][i];
+      let region_end = req.body.regionInfo['end'][i];
+      if(region_chr.concat(':', region_start, '-', region_end) === region){
+	if(req.body.regionInfo['chunk'][i] !== ''){
+	  chunkPath = req.body.regionInfo['chunk'][i]
+	}
+      }
+      i += 1;
+    }
+    // check that the 'chunk.vg' file exists in the chunk folder
+    let chunk_file = `${dataPath}${chunkPath}/chunk.vg`;
```

Comment:
`chunkPath` has been sent in from the client; we definitely can't just paste it into a file path without making sure it doesn't go up a directory.

Reply:
Commit 8db286 now checks that the paths used here and in general are "allowed" which I defined as no backtracking `..` and not starting with `/etc`, `/sys`, `/usr`.

Reply:
I don't think just banning `/etc`, `/sys`, and `/usr` is right. I think the user expectation here is that the tube map server is only going to serve stuff out of the specific configured backend directory/directories, not that it's going to potentially serve everything on the hard drive except those. What if the tube map is supposed to be serving `/home/jmonlong/publicData/tubeMap/HPRC.gbz` and we definitely don't want it to be able to serve `/home/jmonling/privateData/peopleWithCancer/BrettThePatientsEntireGenome.gbz`?

I think we need a collection of the allowed directories, and we need to verify that the path we are looking at is actually in one of them.